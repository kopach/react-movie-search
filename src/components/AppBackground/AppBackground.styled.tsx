import styled from 'styled-components'
import MatrixParallax from 'react-matrix-parallax'

export const BackgroundImage = styled.div`
  background: #2b3e50 url(./background.jpg) no-repeat center 0;
  background-attachment: fixed;
  position: fixed;
  width: 100vw;
  height: 100vh;
  opacity: 0.5;
`

// TODO: move this
export const matrixColor = 'rgba(122, 229, 114, 0.87)'

export const StyledMatrixParallax = styled(MatrixParallax).attrs({
  parallaxRate: 50,
  color: matrixColor,
})``

export const Transparent = styled.div`
  opacity: 0.6;
`
