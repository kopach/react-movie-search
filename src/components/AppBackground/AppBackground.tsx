import React from 'react'
import { BackgroundImage, StyledMatrixParallax, Transparent } from './AppBackground.styled'

export const AppBackground: React.FC = (): JSX.Element => (
  <>
    <BackgroundImage />
    <Transparent>
      <StyledMatrixParallax />
    </Transparent>
  </>
)
