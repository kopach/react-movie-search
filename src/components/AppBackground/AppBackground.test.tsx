import React from 'react'
import { shallow } from 'enzyme'
import { AppBackground } from './AppBackground'

describe('<AppBackground />', () => {
  let component: ReturnType<typeof shallow>

  beforeEach(() => {
    component = shallow(<AppBackground />)
  })

  it('should mount', () => {
    expect(component.length).toBe(1)
  })

  it('renders as expected', () => {
    expect(component).toMatchSnapshot()
  })
})
