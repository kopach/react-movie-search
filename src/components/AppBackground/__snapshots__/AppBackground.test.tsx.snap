// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBackground /> renders as expected 1`] = `
<Fragment>
  <styled.div />
  <styled.div>
    <Styled(WithStyles(MatrixParallax)) />
  </styled.div>
</Fragment>
`;
